### PR TITLE
Cleanup the `actor` package

### PR DESCRIPTION
**Pull request by shsms**
The `frequenz.sdk.actor` package was shared by the `Actor` implementation and by a number of core actors of the SDK and other utilities.

This PR moves all the core actors and utilities into more appropriate locations.

- `frequenz.sdk.`
  + `actor.ChannelRegistry` → `_internal._channels.ChannelRegistry`
  + `actor.ConfigManagingActor` → `config.ConfigManagingActor`
  + `actor._power_managing` → `microgrid._power_managing`
  + `actor.power_distributing` → `microgrid._power_distributing`
  + `actor._resampling` → `microgrid._resampling`
  + `actor._data_sourcing` → `microgrid._data_sourcing`

This is part of #575

### CODE REVIEW

File: src/frequenz/sdk/microgrid/_power_wrapper.py [hunk 1]
```diff
@@ -17,19 +16,13 @@
...
 
 from .._internal._channels import ChannelRegistry, ReceiverFetcher
-from . import _power_managing
-
-# A number of imports had to be done inside functions where they are used, to break
-# import cycles.
-#
-# pylint: disable=import-outside-toplevel
-if typing.TYPE_CHECKING:
-    from ..actor.power_distributing import (  # noqa: F401 (imports used by string type hints)
```
Comment (llucax) [inline src/frequenz/sdk/microgrid/_power_wrapper.py:27]: Do we still have the "string type hints" or was the comment outdated?

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -23,6 +23,8 @@
 
 - The `ConfigManagingActor` now uses `collections.abc.Mapping` as the output sender type. This change indicates that the broadcasted configuration is intended to be read-only.
 
+- The `PowerDistributingActor` is no longer part of the public API.  The `Result` types for the power distribution results are now exposed through the `frequenz.sdk.microgrid.battery_pool` module.
```
Comment (llucax) [inline RELEASE_NOTES.md]: Technically all the other actors in `frequenz.sdk.actor` are not longer part of the public API either, they were exposed at the package level (`frequenz/sdk/actor/__init__.py`).

File: src/frequenz/sdk/timeseries/_voltage_streamer.py
Comment (llucax) [inline src/frequenz/sdk/timeseries/_voltage_streamer.py:1]: Interesting that this module still needs to use `TYPE_CHECKING` and local import for stuff in `microgrid`. Do you remember why this is?

In any case I guess it should be gone when we move microgrid-related stuff from `sdk.timeseries` to `sdk.microgrid` too, right?

**Review by llucax — COMMENTED** [review]
Amazing to see many local imports and `TYPE_CHECKING` gone :tada:

File: src/frequenz/sdk/timeseries/_voltage_streamer.py
Comment (shsms) [reply src/frequenz/sdk/timeseries/_voltage_streamer.py:1]: It shouldn't be required, it wasn't needed for all checks and tests to run.  But pylint on the code example in that source file was failing without the `TYPE_CHECKING` and non-top-level-imports.  Will look into that in next PR when moving it to microgrid.

File: src/frequenz/sdk/microgrid/_power_wrapper.py (hunk 1 above)
Comment (shsms) [reply src/frequenz/sdk/microgrid/_power_wrapper.py:27]: I don't think there were string type hints in that file.  My guess is this was copied over from somewhere when we added flake8.

File: RELEASE_NOTES.md (hunk 2 above)
Comment (shsms) [reply RELEASE_NOTES.md]: looks like I missed a lot of stuff.  Updated.

**Review by llucax — APPROVED** [review]
:tada:

**Pull request merged**